src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseEditPageE2ETest.java

```diff
@@ -0,0 +1,122 @@
+package teammates.e2e.cases.e2e;
+
+import java.time.ZoneId;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.AppPage;
+import teammates.e2e.pageobjects.InstructorCourseEditPage;
+import teammates.e2e.pageobjects.InstructorHomePage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#INSTRUCTOR_COURSE_EDIT_PAGE}.
+ */
+public class InstructorCourseEditPageE2ETest extends BaseE2ETestCase {
+    CourseAttributes course;
+    InstructorAttributes[] instructors = new InstructorAttributes[5];
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorCourseEditPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        course = testData.courses.get("InsCrsEdit.CS2104");
+        instructors[0] = testData.instructors.get("InsCrsEdit.helper");
+        instructors[1] = testData.instructors.get("InsCrsEdit.manager");
+        instructors[2] = testData.instructors.get("InsCrsEdit.observer");
+        instructors[3] = testData.instructors.get("InsCrsEdit.coowner");
+        instructors[4] = testData.instructors.get("InsCrsEdit.tutor");
+    }
+
+    @Test
+    public void testAll() {
+        ______TS("verify cannot edit without privilege");
+        // log in as instructor with no edit privilege
+        AppUrl url = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_EDIT_PAGE)
+                .withUserId(instructors[2].googleId)
+                .withCourseId(course.getId());
+        loginAdminToPage(url, InstructorHomePage.class);
+        InstructorCourseEditPage editPage = AppPage.getNewPageInstance(browser, url, InstructorCourseEditPage.class);
+
+        editPage.verifyCourseNotEditable();
+        editPage.verifyInstructorsNotEditable();
+        editPage.verifyAddInstructorNotAllowed();
+
+        ______TS("verify loaded data");
+        // re-log in as instructor with edit privilege
+        url = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_EDIT_PAGE)
+                .withUserId(instructors[3].googleId)
+                .withCourseId(course.getId());
+        editPage = AppPage.getNewPageInstance(browser, url, InstructorCourseEditPage.class);
+
+        editPage.verifyCourseDetails(course);
+
+        editPage.verifyInstructorDetails(1, instructors[0]);
+        editPage.verifyInstructorDetails(2, instructors[1]);
+        editPage.verifyInstructorDetails(3, instructors[2]);
+        editPage.verifyInstructorDetails(4, instructors[3]);
+        editPage.verifyInstructorDetails(5, instructors[4]);
+
+        ______TS("add instructor");
```

Comment:
In this test case, we add a new instructor and verify that the frontend has the instructor's details. However, the backend is not verified right? We should also check that the database has the new instructor in this course.

Reply:
Added some methods with Backdoor to do some verification for backend